CMakeLists.txt

```diff
@@ -128,7 +123,9 @@ target_link_libraries(${PROJECT_NAME} PUBLIC ${DEP_AWS_LIBS} ${PLATFORM_LIBS})
 if (BYO_CRYPTO)
     target_compile_definitions(${PROJECT_NAME} PRIVATE -DBYO_CRYPTO)
 elseif (NOT WIN32 AND NOT APPLE)
-    target_include_directories(${PROJECT_NAME} PRIVATE $<TARGET_PROPERTY:crypto,INTERFACE_INCLUDE_DIRECTORIES>)
+    if (TARGET crypto)
+        target_include_directories(${PROJECT_NAME} PRIVATE $<TARGET_PROPERTY:crypto,INTERFACE_INCLUDE_DIRECTORIES>)
```

Comment:
this `target_include_directories()` call would make more sense up in the branch where we call `set(PLATFORM_LIBS crypto dl)` 

rather than trying to reproduce the sequence of if/else/if/else again down here

Reply:
I might be able to remove these altogether once the targets are fixed. Let me see if I can get everything to build together with it off and I'll just remove all of these.

Reply:
removed